monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,307 @@
+    """
+    Generate foreground probability map by hover map. The more parts of the image that cannot be identified as foreground areas,
+    the larger the grey scale value. The grey value of the instance's border will be larger.
+
+    Args:
+        kernel_size: the size of the Sobel kernel. Defaults to 21.
+        remove_small_objects: whether need to remove some objects in segmentation results. Defaults to True.
+        min_size: objects smaller than this size are removed if `remove_small_objects` is True. Defaults to 10.
+
+    Raises:
+        ValueError: when the `mask` shape is not [1, H, W].
+        ValueError: when the `hover_map` shape is not [2, H, W].
+
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(self, kernel_size: int = 21, min_size: int = 10, remove_small_objects: bool = True) -> None:
+        self.sobel_gradient = SobelGradients(kernel_size=kernel_size)
+        if remove_small_objects:
+            self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        else:
+            self.remove_small_objects = None  # type: ignore
+
+    def __call__(self, mask: NdarrayOrTensor, hover_map: NdarrayOrTensor) -> NdarrayOrTensor:  # type: ignore
+        """
+        Args:
+            mask: binarized segmentation result.  Shape must be [1, H, W].
+            hover_map:  horizontal and vertical distances of nuclear pixels to their centres of mass. Shape must be [2, H, W].
+                The first and second channel represent the horizontal and vertical maps respectively. For more details refer
+                to papers: https://arxiv.org/abs/1812.06499.
+
+        Return:
+            Foreground probability map.
+        """
+        if len(mask.shape) != 3 or len(hover_map.shape) != 3:
+            raise ValueError(
+                f"Suppose the mask and hover map should be with shape of [C, H, W], but got {mask.shape}, {hover_map.shape}"
+            )
+        if mask.shape[0] != 1:
+            raise ValueError(f"Suppose the mask only has one channel, but got {mask.shape[0]}")
+        if hover_map.shape[0] != 2:
+            raise ValueError(f"Suppose the hover map only has two channels, but got {hover_map.shape[0]}")
+
+        hover_h = hover_map[0:1, ...]
+        hover_v = hover_map[1:2, ...]
+
+        hover_h = (hover_h - min(hover_h)) / (max(hover_h) - min(hover_h))
+        hover_v = (hover_v - min(hover_v)) / (max(hover_v) - min(hover_v))
+        sobelh = self.sobel_gradient(hover_h)[0, ...]
+        sobelv = self.sobel_gradient(hover_v)[1, ...]
+        sobelh = 1 - (sobelh - min(sobelh)) / (max(sobelh) - min(sobelh))
+        sobelv = 1 - (sobelv - min(sobelv)) / (max(sobelv) - min(sobelv))
+
+        # combine the h & v values using max
+        overall = maximum(sobelh, sobelv)
+        overall = overall - (1 - mask)
+        overall[overall < 0] = 0
+
+        return convert_to_dst_type(overall, mask, dtype=np.float32)[0]
+
+
+class GenerateDistanceMap(Transform):
```

Comment:
This part is referred to as energy landscape in the paper: https://arxiv.org/pdf/1812.06499.pdf, so you might want to consider that name too.

Reply:
Hi @drbeh, thanks for your advice, I just checked the original repo and find they use `dist`.
https://github.com/vqdang/hover_net/blob/842964dc5d26ffe196d236d16fad16643a103850/models/hovernet/post_proc.py#L74
I'm not sure which one should I use, but in my simple mind, I think `distance map` is more understandable and also more consistent with the examples in [skimage's docs](https://scikit-image.org/docs/stable/api/skimage.segmentation.html#skimage.segmentation.watershed). What do you think?

Reply:
Thank you @KumoLiu, sure, that's fine too.